[DX] Various typo, grammar, and info fixes (#10433)
hackclub/hcb: dev-docs/guides/reimbursements_transaction_engine.md

Reimbursements are one of the more complicated parts of HCB! This isn’t a comprehensive guide to how they work, instead, this describes how money flows through the system. 

To send the person being reimbursed money, we use standard HCB transfers such as an `AchTransfer`. However, we send these transfers from the “HCB Reimbursements Clearinghouse” organisation. On the organisation that is sending the reimbursement we have one transaction on the ledger per reimbursed expense (these are called `ExpensePayout`s). We took this approach to increase transparency and make it easier to understand what a reimbursement is for.

A `Reimbursement::Report` is a collection of `Reimbursement::Expenses`. No money moves until a report is approved by an admin and if needed, an organiser. 

When a `Reimbursement::Report` is marked as `reimbursement_approved` by an admin, we run `Reimbursement::Report#reimburse!` which creates one `ExpensePayout` per approved expense and one `PayoutHolding` for the report.

> These transactions' internal book transfers move money from the organisation that is reimbursing someone to the clearinghouse organisation.

Reviewer: ```suggestion
These transactions are internal book transfers move money from the organisation that is reimbursing someone to the clearinghouse organisation.
```

> These transactions' internal book transfers move money from the organisation that is reimbursing someone to the clearinghouse organisation.

Reviewer: ```suggestion
These transactions are internal book transfers that move money from the organisation that is reimbursing someone to the clearinghouse organisation.
```

An `ExpensePayout` is a book transfer from “FS Main” to “FS Operating”. It comes in as a negative `CanonicalTransaction`.

A `PayoutHolding` is a book transfer from “FS Operating” to “FS Main”. It comes in as a positive `CanonicalTransaction`. A `Reimbursement::Report` can only have one `PayoutHolding`. Its `amount_cents` will be the sum of the `Reimbursement::Report`’s `ExpensePayout`s’ `amount_cents`.

These are both created in `Reimbursement::Report#reimburse!` 

After they are both created, they have `after_create` callbacks that create a `CanonicalPendingTransaction`. That means there will immediately be a transaction on the ledgers of both the reimbursing organisation and HCB Reimbursements Clearinghouse.

`Reimbursement::ExpensePayoutService::Nightly` is ran every five minutes and creates Column book transfers for each of these. 

> It also marks any `ExpensePayout`s with a `CanonicalTransaction` and any `PayoutHolding`s with either a `CanonicalTransaction` or a `CanonicalPendingTransaction` as settled. You’ll notice that every `PayoutHolding` should meet this condition. This is true. It used to wait for a `CanonicalTransaction` but we started fronting the `CanonicalPendingTransaction` so it would immediately increase HCB Reimbursement Clearinghouse’s balance. That’s because these transactions are guaranteed to end up on our bank statement. 

Reviewer: Please keep the two paragraphs broken up

### Failed `PayoutHolding`s

> If we attempt to send an `AchTransfer` and it fails (this means it errors), we mark the `PayoutHolding` as `failed`.

Reviewer: ```suggestion
If we attempt to send an `AchTransfer` and Column errors or it is returned, we mark the `PayoutHolding` as `failed`.
```

raise ArgumentError, "a check is present" if increase_check.present?
```

> and if it passes, we create a set of Column book transfers that are essentially the reverse of what we created above. 

Reviewer: I prefer the capitalised And
